src/libstd/sys/windows/c.rs

```diff
@@ -1028,34 +1062,39 @@ extern "system" {
     pub fn LeaveCriticalSection(CriticalSection: *mut CRITICAL_SECTION);
     pub fn DeleteCriticalSection(CriticalSection: *mut CRITICAL_SECTION);
 
+    #[cfg(not(target_os = "uwp"))]
     pub fn ReadConsoleW(hConsoleInput: HANDLE,
                         lpBuffer: LPVOID,
                         nNumberOfCharsToRead: DWORD,
                         lpNumberOfCharsRead: LPDWORD,
                         pInputControl: PCONSOLE_READCONSOLE_CONTROL) -> BOOL;
 
+    #[cfg(not(target_os = "uwp"))]
     pub fn WriteConsoleW(hConsoleOutput: HANDLE,
                          lpBuffer: LPCVOID,
                          nNumberOfCharsToWrite: DWORD,
                          lpNumberOfCharsWritten: LPDWORD,
                          lpReserved: LPVOID) -> BOOL;
 
+    #[cfg(not(target_os = "uwp"))]
```

Comment:
This file is one where there's a lot of `#[cfg]`, so it might be good to group these up into a macro? It'd be nice to do something like:

```rust
if_not_uwp! {
    // .. everything that's `#[cfg(not(..))]`
}

if_uwp! {
    // ... everything that's #[cfg]
}
```

Reply:
So... the macro would expand all line within it as `#cfg[not(..)]` but it would appear only as a nice block, correct? I have absolutely 0 experience with rust macros, so I prefer to double check the intent :)

Reply:
You can actually have a generic macro for grouping items which you can then apply `#[cfg]` to.
```rust
macro_rules! ifdef {
    ($($thing:item)*) => ($($thing)*)
}
```
```rust
#[cfg(foo)]
ifdef! {
    // everything
}
```

Reply:
Ah yeah what @retep998 mentioned would work here, and you could even change it to:

```rust
macro_rules! ifdef {
    ($($t:tt)*) => ($($t)*)
}
```

Reply:
That requires to have 3 different `extern "system"` scopes, as far as I can see. Is that ok for you?

Reply:
Yeah I think that's fine as the main thing I'm interested in is reducing the `#[cfg]` traffic